credentialsd-ui/src/gui/view_model/gtk/mod.rs

```diff
@@ -134,34 +134,38 @@ impl ViewModel {
                                     view_model.waiting_for_device(&device)
                                 }
                                 ViewUpdate::UsbNeedsPin { attempts_left } => {
-                                    let prompt = match attempts_left {
-                                        Some(1) => {
-                                            "Enter your PIN. 1 attempt remaining.".to_string()
-                                        }
-                                        Some(attempts_left) => format!(
-                                            "Enter your PIN. {attempts_left} attempts remaining."
-                                        ),
-                                        None => "Enter your PIN.".to_string(),
-                                    };
+                                    let mut prompt = gettext("Enter your PIN.");
+                                    if let Some(left) = attempts_left {
+                                        let localized = ngettext(
+                                            "One attempt remaining.",
+                                            "%d attempts remaining.",
+                                            left,
+                                        );
+                                        prompt.push(' ');
```

Comment:
Concatenation with a space does not work in all locales (eg. neither Chinese or Japanese use spaces to separate sentences AFAIK). This should be a single combined string, or concatenation should happen through a localisable placeholder.